fragment/fragment/src/main/java/androidx/fragment/app/FragmentActivity.java

```diff
@@ -645,6 +649,19 @@ public void startActivityFromFragment(@NonNull Fragment fragment,
         fragment.startActivityForResult(intent, requestCode, options);
     }
 
+    @Override
+    public void reportFullyDrawn() {
+        // On KitKat (API 19) the Activity.reportFullyDrawn() method requires the
+        // UPDATE_DEVICE_STATS permission, otherwise it throws an exception. Instead of throwing,
+        // we fall back to a no-op call here.
+        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT && ContextCompat
+                .checkSelfPermission(this, Manifest.permission.UPDATE_DEVICE_STATS)
+                != PackageManager.PERMISSION_GRANTED) {
+            return;
+        }
+        super.reportFullyDrawn();
```

Comment:
Note that this should be guarded with a `Build.VERSION.SDK_INT >= 19` so that `reportFullyDrawn()` can be called on all API levels.

That means it may be more appropriate to re-arrange this method to be something more like:
```
if (Build.VERSION.SDK_INT >= 20) {
  super.reportFullyDrawn();
} else if (Build.VERSION.SDK_INT == 19 && ContextCompat...) {
  // Your current code and comment re: the specific behavior of API 19
}
```